model/ActivityMonitoringService.php

```diff
@@ -237,31 +234,75 @@ protected function getNumberOfActiveUsers($role = null)
     /**
      * Get list of all the deliveries and number of it's executions in each status
      * Result indexed by delivery Uri
+     * @param $params
      * @return array
      */
-    public function getStatesByDelivery()
+    public function getStatesByDelivery(array $params = [])
     {
-        $deliveryMonitoringService = $this->getServiceManager()->get(DeliveryMonitoringService::SERVICE_ID);
-
         $statusesArray = [];
         foreach ($this->deliveryStatuses as $deliveryStatus) {
             $statusesArray[$deliveryStatus->getUri()] = 0;
         }
 
-        $newResult = [];
-        $newResult[self::FIELD_RETIRED_DELIVERIES] = $statusesArray;
-        $newResult[self::FIELD_RETIRED_DELIVERIES]['label'] = self::LABEL_RETIRED_DELIVERIES;
+        /** @var DeliveryMonitoringService $deliveryMonitoringService */
+        $deliveryMonitoringService = $this->getServiceManager()->get(DeliveryMonitoringService::SERVICE_ID);
+        $deliveryService = DeliveryAssemblyService::singleton();
 
-        $deliveries = DeliveryAssemblyService::singleton()->getAllAssemblies();
+        $limit = isset($params['rows']) ? $params['rows'] : 10;
+        $offset = isset($params['page']) ? ($params['page']-1) * $limit : 0;
+        $deliveries = $deliveryService->getRootClass()->getInstances(true, [
```

Comment:
Pagination is not responsibility of `ActivityMonitoringService`. It's responsibility of `DeliveriesActivityDatatable` class. You have to get array of deliveries in `DeliveriesActivityDatatable` (according to `rows` and `page` values) and pass this array into `ActivityMonitoringService::getStatesByDelivery()`.